Sources/ComposableArchitecture/Reducer/Reducers/OnChange.swift

```diff
@@ -0,0 +1,69 @@
+extension ReducerProtocol {
+  /// Adds a reducer to run when this reducer changes the given value in state.
+  ///
+  /// Use this operator to trigger additional logic when a value changes, like when a
+  /// ``BindingReducer`` makes a deeper change to a struct held in ``BindingState``.
+  ///
+  /// ```swift
+  /// struct Settings: ReducerProtocol {
+  ///   struct State {
+  ///     @BindingState var userSettings: UserSettings
+  ///     // ...
+  ///   }
+  ///
+  ///   enum Action: BindableAction {
+  ///     case binding(BindingAction<State>)
+  ///     // ...
+  ///   }
+  ///
+  ///   var body: some ReducerProtocol<State, Action> {
+  ///     BindingReducer()
+  ///       .onChange(of: \.userSettings.isHapticFeedbackEnabled) { oldValue, newValue in
+  ///         Reduce { state, action in
+  ///           .run { send in
+  ///             // Persist new value...
+  ///           }
+  ///         }
+  ///       }
+  ///   }
+  /// }
+  /// ```
+  ///
+  /// When the value changes, the new version of the closure will be called, so any captured values
+  /// will have their values from the time that the observed value has its new value. The system
+  /// passes the old and new observed values into the closure.
+  ///
+  /// > Note: Take care when applying `onChange(of:)` to a reducer, as it adds an equatable check
+  /// > for every action fed into it. Prefer applying it to leaf nodes, like ``BindingReducer``,
+  /// > against values that are quick to equate.
+  ///
+  /// - Parameters:
+  ///   - toValue: A closure that returns a value from the given state.
+  ///   - reducer: A reducer builder closure to run when the value changes.
+  ///   - oldValue: The old value that failed the comparison check.
+  ///   - newValue: The new value that failed the comparison check.
+  /// - Returns: A reducer that performs the
+  public func onChange<V: Equatable, R: ReducerProtocol>(
+    of toValue: @escaping (State) -> V,
```

Comment:
@mbrandonw Should this be `KeyPath<State, V>` instead to discourage doing too much work in the `toValue` block?